test/test_e2e.rb

```diff
@@ -104,4 +104,14 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
```

Comment:
Let's create a new suite for this test, in which we insert a number of failures more than 1 and a number of successful tests too. This way we can assert that only 1 of those failures were in fact executed, and that there were other jobs to be processed in the queue.

It's impossible to do it with `failing_suite` which contains just 1 failure.

Reply:
Ignore my comment as far as the new suite concerned - `failing_suite` already contains 2 failures and 1 success, so we can use that to add the assertions I described in my comment above.